megamek/src/megamek/common/Compute.java

```diff
@@ -3211,27 +3211,33 @@ public static float getExpectedDamage(Game g, WeaponAttackAction waa,
             }
         }
 
-        if (use_table == true) {
+        if (use_table) {
             if (!(attacker instanceof BattleArmor)) {
                 if (weapon.getLinked() == null) {
                     return 0.0f;
                 }
             }
+
             AmmoType at = null;
-            if ((weapon.getLinked() != null)
+            if (waa.getAmmoId() != -1) {
```

Comment:
What does -1 mean here?  Not linked?  Maybe a constant for readability?

Reply:
Ah, right.  -1 is "Unset", and I believe there is a constant, let me grab that...

Reply:
I have created the constant and replaced the magic number "-1" where appropriate.